torchvision/prototype/transforms/_type_conversion.py

```diff
@@ -21,10 +21,8 @@ def __init__(self, num_categories: int = -1):
 
     def _transform(self, inpt: datapoints.Label, params: Dict[str, Any]) -> datapoints.OneHotLabel:
         num_categories = self.num_categories
-        if num_categories == -1 and inpt.categories is not None:
-            num_categories = len(inpt.categories)
         output = one_hot(inpt.as_subclass(torch.Tensor), num_classes=num_categories)
-        return datapoints.OneHotLabel(output, categories=inpt.categories)
+        return datapoints.OneHotLabel(output)
```

Comment:
This does change the functionality, but not by much. Previously, users could use this transform and it would do the right thing by default if the label had some `categories` attached to it. Now they have to explicitly pass it to the constructor.

Note that the `-1` default is somewhat weird. It is a sentinel to let the torch kernel know to "figure it out". This means it checks all values available in the input and infers the number of categories from them. Obviously, if the input doesn't contain the smallest and largest value, this will give false results.